sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -147,7 +147,7 @@ object DateTimeUtils {
    * Converts milliseconds since epoch to SQLTimestamp.
    */
   def millisToMicros(millis: Long): SQLTimestamp = {
-    MILLISECONDS.toMicros(millis)
+    Math.multiplyExact(millis, MICROS_PER_MILLIS)
```

Comment:
I'm surprised `MILLISECONDS.toMicros` is not overflow-safe. AFAIK we use these utils in many places...

Reply:
Yeh, need to review other places.